### PR TITLE
[new release] grenier (v0.7)

### PR DESCRIPTION
**Pull request by let-def**
A collection of various algorithms in OCaml.

- Project page: <a href="https://github.com/let-def/grenier">https://github.com/let-def/grenier</a>
- Documentation: <a href="https://let-def.github.io/grenier/doc">https://let-def.github.io/grenier/doc</a>

##### CHANGES:

Dune & dune-release port contributed by Rudi Grinberg.
Valmari DFA minimization implementation.

### CODE REVIEW

File: packages/grenier/grenier.0.7/opam [hunk 1]
```diff
@@ -0,0 +1,16 @@
+opam-version: "1.2"
+maintainer: "Frederic Bour <frederic.bour@lakaban.net>"
+authors: "Frederic Bour <frederic.bour@lakaban.net>"
+homepage: "https://github.com/let-def/grenier"
+bug-reports: "https://github.com/let-def/grenier"
+license: "ISC"
+dev-repo: "https://github.com/let-def/grenier.git"
+doc: "https://let-def.github.io/grenier/doc"
+build: [
+  ["jbuilder" "subst" "-n" name] {pinned}
```
Comment (kit-ty-kate) [inline packages/grenier/grenier.0.7/opam]: `-p` should be used instead of `-n`. See #12279

**Comment by let-def** [comment]
Changed as requested. Thanks @kit-ty-kate.

**Comment by kit-ty-kate** [comment]
Mmh something is wrong here. You added an extra layer of directory instead :/

```packages/grenier/grenier.0.7/grenier.0.7/opam```

**Comment by let-def** [comment]
Hmmm, sorry for that.

**Comment by kit-ty-kate** [comment]
Seems like grenier is not compatible with OCaml 4.02
```
#=== ERROR while installing grenier.0.7 =======================================#
# opam-version 1.2.2
# os           linux
# command      jbuilder build -p grenier -j 4
# path         /home/travis/.opam/system/build/grenier.0.7
# compiler     system (4.02.3)
# exit-code    1
# env-file     /home/travis/.opam/system/build/grenier.0.7/grenier-7418-e7966f.env
# stdout-file  /home/travis/.opam/system/build/grenier.0.7/grenier-7418-e7966f.out
# stderr-file  /home/travis/.opam/system/build/grenier.0.7/grenier-7418-e7966f.err
### stderr ###
# ml_physh_set.c:275:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
# [...]
# ml_physh_set.c:280:15: error: redefinition of ‘i’
#    for (size_t i = 0; i < sz; ++i)
#                ^
# ml_physh_set.c:275:15: note: previous definition of ‘i’ was here
#    for (size_t i = 0; i < new_sz; ++i)
#                ^
# ml_physh_set.c:280:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
#    for (size_t i = 0; i < sz; ++i)
#    ^
```

**Comment by let-def** [comment]
Thanks. All green now :) (c99 is only 19 years old, it is better to not use such experimental language features)

**Comment by kit-ty-kate** [comment]
Thanks a lot!

**Pull request merged**